ui/spinner.go

```diff
@@ -0,0 +1,47 @@
+package ui
+
+import (
+  "time"
+
+  "github.com/briandowns/spinner"
+)
+
+type Spinner struct{
+  spinner *spinner.Spinner
+  text string
+}
+
+func NewSpinner(text string) *Spinner { return defUI.NewSpinner(text) }
+
+func (u *UI) NewSpinner(text string) *Spinner {
+  s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+  s.Suffix = " " + text
+  return &Spinner {
+    s,
+    text,
+  }
+}
+
+func (s *Spinner) Start() { defUI.StartSpinner(s) }
```

Comment:
Start/stop should be checking whether or not stdout is attached to a terminal or not. I think, if we're not attached to a terminal, we shouldn't print out the spinners (just make it an empty do-nothing call like several other `ui` functions).